InventoryService/cmd/controllers/item_controller.go

```diff
@@ -0,0 +1,263 @@
+// @Success         200
+// @Failure         400
+// @Router          /ingridient/search [get]
+func (controller *ItemController) GetIngridientByFilter(c *gin.Context) {
+	var filters models.ItemModelFilters
+	filters.AlcoholMax = 1
+
+	err := c.BindQuery(&filters)
+
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusBadRequest, err.Error(), nil)
+		return
+	}
+
+	if filters.AlcoholMin < 0 || filters.AlcoholMax > 1 {
+		resources.ResponseJSON(c, http.StatusBadRequest, resources.InadequateAlcohol, nil)
+		return
+	}
+
+	ingridients, err := controller.Client.Ingridient.
+		Query().
+		Where(
+			ingridient.And(
+				ingridient.NameContainsFold(filters.Name),
+				ingridient.AlcoholGT(filters.AlcoholMin),
+				ingridient.AlcoholLT(filters.AlcoholMax),
+				ingridient.IsDry(filters.IsDry),
+			),
+		).
+		All(c)
+
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusInternalServerError, err.Error(), nil)
+		return
+	}
+
+	resources.ResponseJSON(c, http.StatusOK, "", ingridients)
+}
+
+// @BasePath /api/v1
+
+// AddIngridient godoc
+// @Summary           Adds new ingridient
+// @Description       Add new ingridient to database
+// @Tags              Inventory manipulation
+// @Accept            json
+// @Produce           application/json
+// @Param             item body models.ItemModelCreate true "Item data"
+// @Success           201
+// @Failure           400
+// @Router            /ingridient [post]
+func (controller *ItemController) AddIngridient(c *gin.Context) {
+	jsonData, err := c.GetRawData()
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusBadRequest, err.Error(), nil)
+		return
+	}
+
+	var item models.ItemModelCreate
+	json.Unmarshal(jsonData, &item)
+
+	passedValidation, errorMessage := controller.ValidateNewIngridient(c, item)
+
+	if !passedValidation {
+		resources.ResponseJSON(c, http.StatusBadRequest, errorMessage, nil)
+		return
+	}
+
+	ingridient, err := controller.Client.Ingridient.
+		Create().
+		SetName(item.Name).
+		SetAlcohol(item.Alcohol).
+		SetMeasurmentUnit(ingridient.MeasurmentUnit(item.MeasurmentUnit)).
+		SetIsDry(item.IsDry).
+		Save(c)
+
+	if err != nil {
+		resources.ResponseJSON(c, http.StatusInternalServerError, err.Error(), nil)
+		return
+	}
+
+	resources.ResponseJSON(c, http.StatusCreated, resources.IngridientAdded, ingridient)
+}
+
+// @BasePath /api/v1
+
+// AddIngridient godoc
+// @Summary           Changes ingridient information
+// @Description       Change ingridient in the database by id
+// @Tags              Inventory manipulation
+// @Accept            json
+// @Produce           application/json
+// @Param             item body models.ItemModel true "Item id and it's new data"
+// @Success           202
+// @Failure           400
+// @Failure           404
+// @Router            /ingridient [put]
+func (controller *ItemController) ChangeIngridient(c *gin.Context) {
+	jsonData, err := c.GetRawData()
```

Comment:
This sequence:
1) GetRawData()
2) check err != nil
3) json.Unmarshal(...)
Is almost the same in each method, it can be deduplicated

Reply:
Actually it's only used twice, I think it isn't worth it